### PR TITLE
STYLE: specify encodings when opening files

### PR DESCRIPTION
**Pull request by KetuPatel806**
- [x] closes #52997  (Replace #52997 with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.

### CODE REVIEW

**Comment by MarcoGorelli** [comment]
let's stick to one PR, please don't open a new one which does the same thing

same comments from https://github.com/pandas-dev/pandas/pull/52997, please address them

**Comment by KetuPatel806** [comment]
This is my first PR so I was making mistake next time i was more carefull for making PR

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
getting there 💪 

could you please:
- remove `b/.pre-commit-config.yaml`
- run `pre-commit` on the rest of the files you've modified?
```
pre-commit run --from-ref=upstream/main --to-ref=HEAD --all-files
```

**Review by MarcoGorelli — APPROVED** [review]
looks like this is it, thanks @KetuPatel806 !

File: pandas/_testing/contexts.py [hunk 1]
```diff
@@ -126,7 +126,7 @@ def ensure_clean(
     handle_or_str: str | IO = str(path)
     if return_filelike:
         kwargs.setdefault("mode", "w+b")
-        handle_or_str = open(path, **kwargs)
+        handle_or_str = open(path, **kwargs, encoding="utf-8")
```
Comment (MarcoGorelli) [inline pandas/_testing/contexts.py:129]: apologies, looks like some tests are failing here due to
```python
 TypeError: io.open() got multiple values for keyword argument 'encoding'
```

we may have to do something like
```
handle_or_str = open(path, encoding=kwargs.get('encoding', None), **{key: value for key, value in kwargs.items() if key != 'encoding'})
```
, could you try this please and see if it fixes the failing tests?

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]

File: pandas/_testing/contexts.py (hunk 1 above)
Comment (KetuPatel806) [reply pandas/_testing/contexts.py:129]: Ok,I'll do it

Comment (KetuPatel806) [reply pandas/_testing/contexts.py]: it run the pre-commit command and i'll commit it.

File: pandas/tests/io/xml/test_xml.py [hunk 2]
```diff
@@ -1444,7 +1444,7 @@ def test_file_io_iterparse(datapath, parser, mode):
     filename = datapath("io", "data", "xml", "books.xml")
 
     funcIO = StringIO if mode == "r" else BytesIO
-    with open(filename, mode) as f:
+    with open(filename, mode, encoding="utf-8") as f:
```
Comment (MarcoGorelli) [inline pandas/tests/io/xml/test_xml.py:1447]: this one's causing test failures - could we do `encoding="utf-8" if mode=="r" else None`?

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
getting closer

File: pandas/tests/io/xml/test_xml.py [hunk 3]
```diff
@@ -1444,7 +1444,7 @@ def test_file_io_iterparse(datapath, parser, mode):
     filename = datapath("io", "data", "xml", "books.xml")
 
     funcIO = StringIO if mode == "r" else BytesIO
-    with open(filename, mode) as f:
+    with open(filename, mode, encoding="utf-8" if mode=="r" else None) as f:
```
Comment (MarcoGorelli) [inline pandas/tests/io/xml/test_xml.py:1447]: still gonna have to run `pre-commit` on the files you've modifed though ;)

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
there's still some test failures

it may be worth parametrising over both `mode` and `encoding`. either that, or putting similar `encoding='utf-8' if mode=='r' else None` in the failing tests

File: pandas/tests/io/xml/test_xml.py (hunk 3 above)
Comment (KetuPatel806) [reply pandas/tests/io/xml/test_xml.py]: i was run it. it passed all of the hooks. only black (hook) failed it

File: pandas/tests/io/xml/test_xml.py [hunk 4]
```diff
@@ -1444,7 +1444,11 @@ def test_file_io_iterparse(datapath, parser, mode):
     filename = datapath("io", "data", "xml", "books.xml")
 
     funcIO = StringIO if mode == "r" else BytesIO
-    with open(filename, mode) as f:
+    with open(
+        filename,
+        mode="utf-8" if mode == "r" else None,
```
Comment (MarcoGorelli) [inline pandas/tests/io/xml/test_xml.py:1449]: please check this

Comment (KetuPatel806) [reply pandas/tests/io/xml/test_xml.py:1449]: mode = utf-8' if mode=='r' else None this type??

Comment (KetuPatel806) [reply pandas/tests/io/xml/test_xml.py:1449]: which part i can change it refer it i'll do it

Comment (MarcoGorelli) [reply pandas/tests/io/xml/test_xml.py:1449]: you need to change `encoding`, not `mode`

Comment (KetuPatel806) [reply pandas/tests/io/xml/test_xml.py:1449]: i would just try mode="r"

Comment (MarcoGorelli) [reply pandas/tests/io/xml/test_xml.py:1449]: just keep mode as it is

**Comment by MarcoGorelli** [comment]
this is close, a few tests left. would be great if you could finish it up, else I can add a commit on Tuesday and push this over the finish line

**Comment by KetuPatel806** [comment]
i want to finish it up that  test cases

**Comment by MarcoGorelli** [comment]
feel free to fix

**Comment by KetuPatel806** [comment]
can you refer that test cases so i'll fix it?

**Comment by MarcoGorelli** [comment]
check the failing tests in the CI logs

**Comment by KetuPatel806** [comment]
i have check the circle ci to find the error and i tired on a local to fix it.

**Comment by KetuPatel806** [comment]
i try to fix read Binary mode error but if i remove the encoding then pre-commit command does not work it give me error massage called as "unspecified encoding"

**Comment by KetuPatel806** [comment]
i thought it finish it up.

**Review by MarcoGorelli — APPROVED** [review]
should be good, thanks @KetuPatel806 !

**Pull request merged**